### DISCUSSION TITLE
[Suggestion] New tutorial for setting up a custom server

### DISCUSSION THREAD

**Discussion by tippl** [Syncplay/syncplay#445] [Ideas]
The support page has a tutorial on setting up a server, however it is from 2016 and many things have changed.

For example, introduction of TLS support, the new server package and more importantly Debian and Ubuntu switching to the systemd init system as opposed to previous SysVinit. These changes greatly simplify the setup, but also make the old tutorial incompatible with newer installs.

If desirable, i would be willing to writeup a new guide to put on the wiki here possibly, that includes the whole setup.

**Comment by daniel-123** [comment]
I was already planning to write a server guide. Though thinking about it, it might be worth while to bounce some ideas around. I wanted to include the following in the guide:

1. Detailed installation on Debian and its derivatives (deb package) with all commands.
2. Basics of installing it somewhere else (satisfying python requirements, maybe examples for docker?)
3. Detail all the cmd switches, what they mean and whether you should use them.
4. How to make a systemd service unit with it.
5. Maybe include an example how to make TLS work with letsencrypt?

Do you think the guide should include something more or less than the stuff above?

**Comment by tippl** [comment]
Yeah, this is essentially what i was planning to do, with an addition of installing somewhere else.
Description of the command line switches is already in the docs, so the guide probably wouldn't need to go that into detail on them though.

I already have essentially the full install done on my server with following:
- Install on Ubuntu from the server package
- Systemd unit file
- DNS and letsencrypt certificate
- All running under a system user syncplay with explicit ACL to allow only syncplay user access to read the certs

The install is a few months old, so my memory is a bit rusty, but i have all the commands i used in bash history.
